WarshipGame/Assets/Scripts/Gameplay/Grid/HexGrid.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// This class contains all the data of the grid, position of tiles [Vector2Int Coordinates]
+/// </summary>
+public class HexGrid : MonoBehaviour
+{
+    private Dictionary<Vector2Int, HexData> _hexTileDict = new Dictionary<Vector2Int, HexData>();
+    private Dictionary<Vector2Int, List<Vector2Int>> _hexTileNeighboursDict = new Dictionary<Vector2Int, List<Vector2Int>>();
+
+    private HexCoordinates _hexCoords;
+    
+    
+    //fills dictionary with hextiles
+    private void Start()
+    {
+        _hexCoords = FindObjectOfType<HexCoordinates>();
+        
+        foreach (HexData hex in FindObjectsOfType<HexData>())
+        {
+            _hexTileDict[hex.Grid] = hex;
+        }
+    }
+
+    /// <summary>
+    /// Returns the coordinate value of a tile
+    /// </summary>
+    /// <param name="hexCoordinates"></param>
+    /// <returns></returns>
+    public HexData GetTileAt(Vector2Int hexCoordinates)
+    {
+        HexData result = null;
+        _hexTileDict.TryGetValue(hexCoordinates, out result);
```

Comment:
`HexData result = null;` is redudant and can be merged into `TryGetValue()` by doining `out HexData result`